srcs/cpp/src/torch/ops/cuda/collective.cpp

```diff
@@ -76,4 +76,17 @@ int broadcast_cuda_async(torch::Tensor input, torch::Tensor output,
     });
     return handle;
 }
+
+void all_gather_cuda(torch::Tensor input, torch::Tensor output,
+                     const std::string &type)
+{
+    const KungFu_Datatype dtype = from(_torch_tensor_types.at(type));
+    const int np = _default_peer->Size();
+    std::vector<char> send_buffer(data_size(input));
+    std::vector<char> receive_buffer(data_size(input) * np);
```

Comment:
no need to call ` const int np = _default_peer->Size();` here:
`std::vector<char> receive_buffer(data_size(output));`